js/src/components/google-combo-account-card/indicator.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import ConnectedIconLabel from '.~/components/connected-icon-label';
+import LoadingLabel from '.~/components/loading-label/loading-label';
+import useGoogleMCAccount from '.~/hooks/useGoogleMCAccount';
+import useGoogleAdsAccountReady from '.~/hooks/useGoogleAdsAccountReady';
+
+/**
+ * Account creation indicator.
+ * Displays a loading indicator when accounts are being created or a connected icon when accounts are connected.
+ * @param {Object} props Component props.
+ * @param {boolean} props.showSpinner Whether to display a spinner.
+ * @return {JSX.Element|null} Indicator component.
+ */
+const Indicator = ( { showSpinner } ) => {
+	const { googleMCAccount, isPreconditionReady } = useGoogleMCAccount();
+	const isGoogleAdsConnected = useGoogleAdsAccountReady();
+
+	const isGoogleMCConnected =
+		isPreconditionReady &&
+		( googleMCAccount?.status === 'connected' ||
+			( googleMCAccount?.status === 'incomplete' &&
+				googleMCAccount?.step === 'link_ads' ) );
```

Comment:
This same logic is used in the `SetupAccounts` component [here](https://github.com/woocommerce/google-listings-and-ads/blob/7937784777ccfc7cb16ef93f0230ca3231435cd9/js/src/setup-mc/setup-stepper/setup-accounts/index.js#L115-L119).

I suggest we abstract this, similar to how the `useGoogleAdsAccountReady()` hook is written and update the code here and in the `SetupAccount` component to make use of the new hook.

In both cases, I don't think that the `isPreconditionReady` check is necessary. In the previous flow that was mainly used to indicate that the standalone Ads and MC cards were disabled until the Google account was connected.

Reply:
Created a new hook `useGoogleMCAccountReady`: https://github.com/woocommerce/google-listings-and-ads/blob/2859cf56aeae23067101b4675703310ce711d19d/js/src/hooks/useGoogleMCAccountReady.js#L7-L17

Reply:
Nice. Resolving and leaving follow-up feedback below.